diff --git a/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/RNWidget.java b/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/RNWidget.java
index 1c12d1f..4bc736c 100644
--- a/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/RNWidget.java
+++ b/node_modules/react-native-android-widget/android/src/main/java/com/reactnativeandroidwidget/RNWidget.java
@@ -74,8 +74,7 @@ public class RNWidget {
         );
 
         Bitmap bitmap = drawViewToBitmap(widgetWithViews.getRootView());
-        Uri bitmapUri = saveBitmapToDisk(widgetId, "light", bitmap);
-        remoteWidgetView.setImageViewUri(R.id.rn_widget_image_light, bitmapUri);
+        remoteWidgetView.setImageViewBitmap(R.id.rn_widget_image_light, bitmap);
 
         if (widgetWithViews.getRootAccessibilityLabel() != null) {
             remoteWidgetView.setContentDescription(R.id.rn_widget_clickable_container, widgetWithViews.getRootAccessibilityLabel());
@@ -90,8 +89,7 @@ public class RNWidget {
             );
 
             Bitmap darkBitmap = drawViewToBitmap(darkWidgetWithViews.getRootView());
-            Uri darkBitmapUri = saveBitmapToDisk(widgetId, "dark", darkBitmap);
-            remoteWidgetView.setImageViewUri(R.id.rn_widget_image_dark, darkBitmapUri);
+            remoteWidgetView.setImageViewBitmap(R.id.rn_widget_image_dark, darkBitmap);
 
             List<CollectionView> lightCollectionViews = widgetWithViews.getCollectionViews();
             List<CollectionView> darkCollectionViews = darkWidgetWithViews.getCollectionViews();
@@ -101,7 +99,7 @@ public class RNWidget {
                 RNWidgetCollectionService.storeCollection(appContext, widgetId, i, darkCollectionViews.get(i).getRenderedViews(), "dark");
             }
         } else {
-            remoteWidgetView.setImageViewUri(R.id.rn_widget_image_dark, bitmapUri);
+            remoteWidgetView.setImageViewBitmap(R.id.rn_widget_image_dark, bitmap);
 
             List<CollectionView> collectionViews = widgetWithViews.getCollectionViews();
             for (int i = 0; i < Math.min(collectionViews.size(), RNWidgetCollectionService.MAX_COLLECTION_WIDGETS); i++) {
@@ -145,12 +142,6 @@ public class RNWidget {
         return bitmap;
     }
 
-    private Uri saveBitmapToDisk(int widgetId, String mode, Bitmap bitmap) {
-        String fileName = "widget_" + widgetId + "_mode_" + mode + ".png";
-        RNWidgetImageProvider.writeImage(appContext, fileName, bitmap);
-        return RNWidgetImageProvider.getImageUri(appContext, fileName);
-    }
-
     @RequiresApi(Build.VERSION_CODES.N)
     private void addClickableAreas(int widgetId, RemoteViews remoteWidgetView, WidgetWithViews widgetWithViews) {
         remoteWidgetView.removeAllViews(R.id.rn_widget_clickable_container);
